clients/src/main/java/org/apache/kafka/common/errors/NoOffsetForPartitionException.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+/**
+ * Indicates that there is no stored offset for a partition and no defined offset
+ * reset policy.
+ */
+public class NoOffsetForPartitionException extends InvalidOffsetException {
```

Comment:
This exception isn't actually used anywhere. There's an equivalent one in `o.a.k.clients.consumer` which is used. Did we want this to be available under `common` or is under `clients` enough? If the latter, should we clear this one out from this patch? I remember there was discussion of this exception earlier wrt import-control settings, but I don't know if there was a reason it ended up quite like this. We have the same duplication for `OffsetOutOfRangeException`, except that one appears in `Errors.java`.